src/Centreon/Domain/HostConfiguration/Model/HostSeverity.php

```diff
@@ -0,0 +1,231 @@
+    private $alias;
+
+    /**
+     * @var int Priority.
+     */
+    private $level;
+
+    /**
+     * @var Image Define the image that should be associated with this severity.
+     */
+    private $icon;
+
+    /**
+     * @var string|null Comments regarding this severity.
+     */
+    private $comments;
+
+    /**
+     * @var bool Indicates whether this host severity is enabled or not (TRUE by default)
+     */
+    private $isActivated = true;
+
+    /**
+     * @param string $name
+     * @param string $alias
+     * @param int $level
+     * @param Image $icon
+     * @throws \Assert\AssertionFailedException
+     */
+    public function __construct(string $name, string $alias, int $level, Image $icon)
+    {
+        $this->setName($name);
+        $this->setAlias($alias);
+        $this->setLevel($level);
+        $this->setIcon($icon);
+    }
+
+    /**
+     * @return int
+     */
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     * @return HostSeverity
+     */
+    public function setId(int $id): HostSeverity
+    {
+        $this->id = $id;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     * @return HostSeverity
+     * @throws \Assert\AssertionFailedException
+     */
+    public function setName(string $name): HostSeverity
+    {
+        if ($name !== null) {
+            Assertion::maxLength($name, self::MAX_NAME_LENGTH, 'HostSeverity::name');
+        }
+        $this->name = $name;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getAlias(): string
+    {
+        return $this->alias;
+    }
+
+    /**
+     * @param string $alias
+     * @return HostSeverity
+     * @throws \Assert\AssertionFailedException
+     */
+    public function setAlias(string $alias): HostSeverity
+    {
+        if ($alias !== null) {
+            Assertion::maxLength($alias, self::MAX_ALIAS_LENGTH, 'HostSeverity::alias');
+        }
+        $this->alias = $alias;
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isActivated(): bool
+    {
+        return $this->isActivated;
+    }
+
+    /**
+     * @param bool $isActivated
+     * @return HostSeverity
+     */
+    public function setIsActivated(bool $isActivated): HostSeverity
+    {
+        $this->isActivated = $isActivated;
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getLevel(): int
+    {
+        return $this->level;
+    }
+
+    /**
+     * @param int $level
+     * @return HostSeverity
+     * @throws \Assert\AssertionFailedException
+     */
+    public function setLevel(int $level): HostSeverity
+    {
+        if ($level !== null) {
+            Assertion::maxLength((string) $level, self::MAX_LEVEL_LENGTH, 'HostSeverity::level');
```

Comment:
Don't use `maxLength` assert, use `max` instead.
Don't forget to add the `min` assert also